src/gui/mzroll/background_peaks_update.cpp

```diff
@@ -637,10 +624,61 @@ void BackgroundPeakUpdate::setRunFunction(QString functionName) {
 }
 
 void BackgroundPeakUpdate::pullIsotopes(PeakGroup* parentgroup) {
-        peakDetector.pullIsotopes(parentgroup);
+	bool C13Flag = mavenParameters->C13Labeled_BPE;
+	bool N15Flag = mavenParameters->N15Labeled_BPE;
+	bool S34Flag = mavenParameters->S34Labeled_BPE;
+	bool D2Flag = mavenParameters->D2Labeled_BPE;
+
+        IsotopeDetection::IsotopeDetectionType isoType;
+        isoType = IsotopeDetection::PeakDetection;
+
+	IsotopeDetection isotopeDetection(
+                mavenParameters,
+                isoType,
+                C13Flag,
+                N15Flag,
+                S34Flag,
+                D2Flag);
+	isotopeDetection.pullIsotopes(parentgroup);
 }
+
+void BackgroundPeakUpdate::pullIsotopesIsoWidget(PeakGroup* parentgroup) {
```

Comment:
same as pull isotopes(),no ? the only difference is in IsotopeDetectionType

Reply:
@rish9511 It's calling the same isotopeDetection::pullIsotopes() but with different flags and isoType

Reply:
@shubhra-agrawal  Just edited my comment

Reply:
@rish9511 @shubhra-agrawal Ideally, we won't be passing any IsoType to the class. 
There will be only one pullIsotopes function and it will only update one vector of isotopic children (Here we are updating 3 vectors depending on the widget used i.e children, childrenIsoWidget & childrenBarPlot). 

**Why we are updating 3 vectors separately?**
Currently, we are passing a pointer to the class PeakGroup. So updating one vector (children in this case) can lead to errors because 3 widgets will try to update the same vector.

**Solution**
Pass a copy of PeakGroup to every widget rather than passing the pointer.

Reply:
@sahil let's leave it for now then